### ISSUE TITLE
rehaul dependencies

### ISSUE THREAD

**Issue by NiklasPhabian** [SpatioTemporal/STAREPandas#70]
1. I forgot why numpy <1.20 is dependency. Something must have been broken, which seems to work now with numpy 1.22
2. dask should be an optional dependency only if parallelism is wanted
3. pyhdf, h5py, netCDF should be optional dependency only required if granules are read
4. boto3 should be optional dependency only for s3 access
5. matplotlib and astropy are only needed for tests and demos

**Comment by NiklasPhabian** [comment]
As soon as some issues with it might occur, we might also drop support for python 3.6.

**Comment by NiklasPhabian** [comment]
point 2.-5. are really cosmetics as long as they can be installed as easily from pip as they are right now. But we should watch point 1. Pegging an old version is surely not nice.

**Comment by NiklasPhabian** [comment]
```shell
pip3 install numpy==1.20.0
pip3 install pystare==0.8.4
````

```python
import pystare
> ImportError: numpy.core.multiarray failed to import
```

**Comment by NiklasPhabian** [comment]
issue does not appear with numpy>=1.21.0. 
Updating requirements here and in pystare

**Comment by NiklasPhabian** [comment]
I am now starting to think that we pinned numpy because of pandas and geopandas, causing casting warnings.
Many of those seem to be resolved with geopandas>=0.10, but pandas 1.3.5 still seems to trigger some. I'd assume this will get resolved soon.

```
DeprecationWarning: An exception was ignored while fetching the attribute `__array_interface__` from an object of type 'Polygon'.  With the exception of `AttributeError` NumPy will always raise this exception in the future.  Raise this deprecation warning to see the original exception. (Warning added NumPy 1.21)
    result[:] = values
```

**Comment by NiklasPhabian** [comment]
ok. I am closing this regarding point 1 and open an issue for points 2-5
